Service/GmapsAutocompleteService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Daften\Bundle\AddressingBundle\Service;
+
+use CommerceGuys\Addressing\Repository\AddressFormatRepositoryInterface;
+use CommerceGuys\Addressing\Repository\SubdivisionRepositoryInterface;
+use CommerceGuys\Intl\Country\CountryRepositoryInterface;
+use Daften\Bundle\AddressingBundle\Entity\AddressEmbeddable;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Class GmapsAutocompleteService.
+ *
+ * Provides the possibility to inject the GMaps API key and the toString method for an address to prepopulate an
+ * autocomplete address field.
+ */
+class GmapsAutocompleteService
+{
+    /**
+     * @var CountryRepositoryInterface
+     */
+    protected $countryRepository;
+
+    /**
+     * @var AddressFormatRepositoryInterface
+     */
+    protected $addressFormatRepository;
+
+    /**
+     * @var SubdivisionRepositoryInterface
+     */
+    protected $subdivisionRepository;
+
+    /**
+     * @var string
+     */
+    protected $gmapsApiKey;
+
+    public function __construct(
+        CountryRepositoryInterface $countryRepository,
+        AddressFormatRepositoryInterface $addressFormatRepository,
+        SubdivisionRepositoryInterface $subdivisionRepository,
+        string $gmapsApiKey
+    ) {
+        $this->countryRepository = $countryRepository;
+        $this->addressFormatRepository = $addressFormatRepository;
+        $this->subdivisionRepository = $subdivisionRepository;
+        $this->gmapsApiKey = $gmapsApiKey;
+    }
+
+    /**
+     * @return string
+     */
+    public function getGmapsApiKey(): string
+    {
+        return $this->gmapsApiKey;
+    }
+
+    /**
+     * @param string $gmapsApiKey
+     */
+    public function setGmapsApiKey(string $gmapsApiKey): void
+    {
+        $this->gmapsApiKey = $gmapsApiKey;
+    }
+
+    public function addressAutocompleteDefault(AddressEmbeddable $address) {
```

Comment:
What does this do? It doesn't return or set any value?

Reply:
To get the default address format or any address formatting, we should use services IMO. Otherwise we aggregate fields in the form itself, which seems less clean.

Later on, I'd put this function in another more generic service that can be used to format addresses in general (single-line for gmaps; multi-line, ...)

Reply:
Ok, but isn't it missing a `return` statement then?

Reply:
Correct, how I didn't see that. It worked at some point, so must not have committed a change.